manager/state/raft/storage.go

```diff
@@ -69,6 +83,62 @@ func (n *Node) loadAndStart(ctx context.Context, forceNewCluster bool) error {
 	return nil
 }
 
+func migrateWALs(legacyWALDir, walDir string) error {
+	// keep temporary wal directory so WAL initialization appears atomic
+	tmpdirpath := filepath.Clean(walDir) + ".tmp"
+	if fileutil.Exist(tmpdirpath) {
+		if err := os.RemoveAll(tmpdirpath); err != nil {
+			return fmt.Errorf("could not remove temporary wal directory: %v", err)
+		}
+	}
+	if err := fileutil.CreateDirAll(tmpdirpath); err != nil {
+		return fmt.Errorf("could not create temporary wal directory: %v", err)
+	}
+
+	walNames, err := fileutil.ReadDir(legacyWALDir)
+	if err != nil {
+		return fmt.Errorf("could not list WAL directory %s: %v", legacyWALDir, err)
+	}
+
+	for _, fname := range walNames {
+		_, err := copyFile(filepath.Join(legacyWALDir, fname), filepath.Join(tmpdirpath, fname), 0600)
+		if err != nil {
+			return fmt.Errorf("error copying WAL file: %v", err)
+		}
+	}
+
+	if err := os.Rename(tmpdirpath, walDir); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// copyFile copies from src to dst until either EOF is reached
```

Comment:
It is copying to a temporary directory that gets renamed at the end, so I think copying in place is okay.